### ISSUE TITLE
Compilation on OSX Big Sur, gcc=10.2, gfortran 10.2

### ISSUE THREAD

**Issue by ungur** [xianyi/OpenBLAS#2991]
I had the following issue when compiling the OpenBLAS version  v.0.3.9. on my laptop, OSX, Big Sur, using the GCC and GFORTRAN 10.2.

The issue happens when LAPACKE is being compiled.  Setting the NO_LAPACKE=1 in Makefile.rule allows a successful finishing of the OpenBLAS compilation.

```


lapacke_cggsvp_work.c:52:9: error: implicit declaration of function 'cggsvp_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LAPACK_cggsvp( &jobu, &jobv, &jobq, &m, &p, &n, a, &lda, b, &ldb, &tola,
        ^
../include/lapack.h:3769:23: note: expanded from macro 'LAPACK_cggsvp'
#define LAPACK_cggsvp LAPACK_GLOBAL(cggsvp,CGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:241:1: note: expanded from here
cggsvp_
^
lapacke_cggsvp_work.c:52:9: note: did you mean 'cggsvp3_'?
../include/lapack.h:3769:23: note: expanded from macro 'LAPACK_cggsvp'
#define LAPACK_cggsvp LAPACK_GLOBAL(cggsvp,CGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:241:1: note: expanded from here
cggsvp_
^
../include/lapack.h:3791:6: note: 'cggsvp3_' declared here
void LAPACK_cggsvp3(
     ^
../include/lapack.h:3790:24: note: expanded from macro 'LAPACK_cggsvp3'
#define LAPACK_cggsvp3 LAPACK_GLOBAL(cggsvp3,CGGSVP3)
                       ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:108:1: note: expanded from here
cggsvp3_
^
lapacke_cggsvp_work.c:139:9: error: implicit declaration of function 'cggsvp_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LAPACK_cggsvp( &jobu, &jobv, &jobq, &m, &p, &n, a_t, &lda_t, b_t,
        ^
../include/lapack.h:3769:23: note: expanded from macro 'LAPACK_cggsvp'
#define LAPACK_cggsvp LAPACK_GLOBAL(cggsvp,CGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:242:1: note: expanded from here
cggsvp_
^
2 errors generated.
make[3]: *** [lapacke_cggsvp_work.o] Error 1
make[3]: *** Waiting for unfinished jobs....
lapacke_dggsvp_work.c:48:9: error: implicit declaration of function 'dggsvp_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LAPACK_dggsvp( &jobu, &jobv, &jobq, &m, &p, &n, a, &lda, b, &ldb, &tola,
        ^
../include/lapack.h:3761:23: note: expanded from macro 'LAPACK_dggsvp'
#define LAPACK_dggsvp LAPACK_GLOBAL(dggsvp,DGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:241:1: note: expanded from here
dggsvp_
^
lapacke_dggsvp_work.c:48:9: note: did you mean 'dggsvp3_'?
../include/lapack.h:3761:23: note: expanded from macro 'LAPACK_dggsvp'
#define LAPACK_dggsvp LAPACK_GLOBAL(dggsvp,DGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:241:1: note: expanded from here
dggsvp_
^
../include/lapack.h:3808:6: note: 'dggsvp3_' declared here
void LAPACK_dggsvp3(
     ^
../include/lapack.h:3807:24: note: expanded from macro 'LAPACK_dggsvp3'
#define LAPACK_dggsvp3 LAPACK_GLOBAL(dggsvp3,DGGSVP3)
                       ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:109:1: note: expanded from here
dggsvp3_
^
lapacke_dggsvp_work.c:127:9: error: implicit declaration of function 'dggsvp_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LAPACK_dggsvp( &jobu, &jobv, &jobq, &m, &p, &n, a_t, &lda_t, b_t,
        ^
../include/lapack.h:3761:23: note: expanded from macro 'LAPACK_dggsvp'
#define LAPACK_dggsvp LAPACK_GLOBAL(dggsvp,DGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:242:1: note: expanded from here
dggsvp_
^
2 errors generated.
make[3]: *** [lapacke_dggsvp_work.o] Error 1
lapacke_zggsvp_work.c:52:9: error: implicit declaration of function 'zggsvp_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LAPACK_zggsvp( &jobu, &jobv, &jobq, &m, &p, &n, a, &lda, b, &ldb, &tola,
        ^
lapacke_sggsvp_work.c:48:9: error: implicit declaration of function 'sggsvp_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LAPACK_sggsvp( &jobu, &jobv, &jobq, &m, &p, &n, a, &lda, b, &ldb, &tola,
        ^
../include/lapack.h:3779:23: note: expanded from macro 'LAPACK_zggsvp'
#define LAPACK_zggsvp LAPACK_GLOBAL(zggsvp,ZGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:241:1: note: expanded from here
zggsvp_
^
../include/lapack.h:3753:23: note: expanded from macro 'LAPACK_sggsvp'
#define LAPACK_sggsvp LAPACK_GLOBAL(sggsvp,SGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:241:1: note: expanded from here
sggsvp_
^
lapacke_zggsvp_work.c:52:9: note: did you mean 'zggsvp3_'?
../include/lapack.h:3779:23: note: expanded from macro 'LAPACK_zggsvp'
#define LAPACK_zggsvp LAPACK_GLOBAL(zggsvp,ZGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:241:1: note: expanded from here
zggsvp_
^
../include/lapack.h:3840:6: note: 'zggsvp3_' declared here
void LAPACK_zggsvp3(
     ^
../include/lapack.h:3839:24: note: expanded from macro 'LAPACK_zggsvp3'
lapacke_sggsvp_work.c:48:9: #define LAPACK_zggsvp3 LAPACK_GLOBAL(zggsvp3,ZGGSVP3)
note                       ^: 
did you../include/lapacke_mangling.h :mean12: 39'sggsvp3_'?: 
note: expanded ../include/lapack.hfrom: 3753macro: 23'LAPACK_GLOBAL': 
note: expanded from#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_ 
macro                                      ^ 
'LAPACK_sggsvp'
<scratch space>:111:1: note: expanded from here
zggsvp3_
^
#define LAPACK_sggsvp LAPACK_GLOBAL(sggsvp,SGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:241:1: note: expanded from here
sggsvp_
^
../include/lapack.h:3824:6: note: 'sggsvp3_' declared here
void LAPACK_sggsvp3(
     ^
../include/lapack.h:3823:24: note: expanded from macro 'LAPACK_sggsvp3'
#define LAPACK_sggsvp3 LAPACK_GLOBAL(sggsvp3,SGGSVP3)
                       ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:110:1: note: expanded from here
sggsvp3_
^
lapacke_zggsvp_work.c:139:9: error: implicit declaration of function 'zggsvp_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LAPACK_zggsvp( &jobu, &jobv, &jobq, &m, &p, &n, a_t, &lda_t, b_t,
        ^
../include/lapack.h:3779:23: note: expanded from macro 'LAPACK_zggsvp'
#define LAPACK_zggsvp LAPACK_GLOBAL(zggsvp,ZGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:242:1: note: expanded from here
zggsvp_
^
lapacke_sggsvp_work.c:127:9: error: implicit declaration of function 'sggsvp_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LAPACK_sggsvp( &jobu, &jobv, &jobq, &m, &p, &n, a_t, &lda_t, b_t,
        ^
../include/lapack.h:3753:23: note: expanded from macro 'LAPACK_sggsvp'
#define LAPACK_sggsvp LAPACK_GLOBAL(sggsvp,SGGSVP)
                      ^
../include/lapacke_mangling.h:12:39: note: expanded from macro 'LAPACK_GLOBAL'
#define LAPACK_GLOBAL(lcname,UCNAME)  lcname##_
                                      ^
<scratch space>:242:1: note: expanded from here
sggsvp_
^
2 errors generated.
2 errors generated.
make[3]: *** [lapacke_zggsvp_work.o] Error 1
make[3]: *** [lapacke_sggsvp_work.o] Error 1
make[2]: *** [lapacke] Error 2
make[1]: *** [lapackelib] Error 2
make: *** [netlib] Error 2


```

**Comment by martin-frbg** [comment]
This was fixed by #2808 a while ago, the patch should apply cleanly to 0.3.9 if you do not want to update

**Comment by ungur** [comment]
Thanks, Martin!  My mistake as I missed the later tags... The standard compilation of v0.3.12 works fine.
